sdk/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsClient.cs

```diff
@@ -328,7 +328,14 @@ public virtual async Task<Response<DetectLanguageResultCollection>> DetectLangua
                 {
                     case 200:
                         IDictionary<string, int> map = CreateIdToIndexMap(documents);
-                        return await CreateDetectLanguageResponseAsync(response, map, cancellationToken).ConfigureAwait(false);
+                        DetectLanguageResultCollection results = await CreateDetectLanguageResponseAsync(response, map, cancellationToken).ConfigureAwait(false);
+                        if (results[0].HasError && string.IsNullOrEmpty(results[0].Id))
```

Comment:
Would it be better to just check if id == "", since id == null means something else

Reply:
+1